sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -145,6 +147,16 @@ export class ApplicationService extends RecordDataModelService<Application> {
     originalAssessment.submittedBy = auditUser;
     originalAssessment.submittedDate = now;
     originalAssessment.creator = auditUser;
+    const institutionLocation =
```

Comment:
I am wondering should this logic be only checked if `applicationData.selectedLocation,` is there or can be moved after the draft logic. because in draft state, a student does have the option to omit the `program` tab and fill other tabs right? And if they do so, this logic will not allow the student to submit the draft application right?

Reply:
And as per the discussion, I believe we decided not to add location id in the draft state right? unless I am missing something https://github.com/bcgov/SIMS/pull/1887#discussion_r1167328352. so , do we need line 201 and this logic we can move after the draft logic right?

Reply:
@ann-aot adding the location for a draft application was never on the table. The code on line 201 is inside the block that converts a draft to a submitted application.
![image](https://user-images.githubusercontent.com/61259237/232839936-32575066-ba00-4173-be93-8b254fd00be8.png)

Reply:
oh.. ya... Thanks for that. resolving the comment